### DISCUSSION TITLE
Dealing with future builders / uninitialized form fields

### DISCUSSION THREAD

**Discussion by Adam-Langley** [danvick/flutter_form_builder#1366] [General]
I am using Skeletonizer:
https://pub.dev/packages/skeletonizer

This means that while I'm loading my form data from a repository, using futures/streams, I will still render my entire form, but wrapped in a Skeletonizer and populated with dummy data.

This causes a problem that all the form elements are hence first built with 'initialValue' of some irrelevent value (such as null). Then, when the future completes, and the form data is loaded - it rebuilds again and nothing really rebuilds - all the form elements are still empty.

One option is to build the form fields with some "magic" initialValue - however this is terrible, it will completely break the 'isDirty' flag.

What's the solution here?
I really need a way to force a Form Field to "re-initialize" once the form data goes from 'loading' to 'loaded'... ideas?

**Comment by deandreamatias** [comment]
Hi @Adam-Langley 
Sorry for delay.
You can solve this? Maybe can use `_formKey.currentState?.patchValue(mapFormValues)` to load data on form
